src/pages/Editor/Classes/UpdateClassModals/EditClassModal.tsx

```diff
@@ -0,0 +1,165 @@
+  });
+  const { fields, append, remove } = useFieldArray({
+    name: 'classworks',
+    control,
+  });
+  useEffect(() => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+  }, [item.name, item.description, setValue]);
+
+  const [errorMessages, setErrorMessages] = useState<ErrorMessageInterface[]>(
+    []
+  );
+
+  // ClassWork
+
+  const handleCancelModal = () => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+    reset();
+    toggle();
+  };
+
+  const onSubmit = async (data: CreateClassProps) => {
+    try {
+      await editClass(data, item._id);
+      fetchClasses();
+      reset();
+      toggle();
+    } catch (error) {
+      setErrorMessages(error as ErrorMessageInterface[]);
+    }
+  };
+
+  return (
+    <Modal
+      titleText="Edit Class"
+      isOpen={isOpen}
+      onCloseModal={toggle}
+      primaryAction={
+        <PrimaryButton type="submit" form="EditClassForm">
+          Submit
+        </PrimaryButton>
+      }
+      secondaryAction={<Button onClick={handleCancelModal}>Cancel</Button>}
+      customStyles={{
+        content: {
+          height: '90vh',
+          width: '90vw',
+        },
+      }}
+    >
+      <StyledForm id="EditClassForm" onSubmit={handleSubmit(onSubmit)}>
+        <InputField htmlFor="name">
+          <span>Class Title</span>
+          <Input
+            type="text"
+            id="name"
+            placeholder="Enter Class Title"
+            {...register('name', classValidationRules.name)}
+          />
+        </InputField>
+        <InputField htmlFor="description">
+          <span>Class Description</span>
+          <TextArea
+            id="description"
+            placeholder="Enter Class description"
+            {...register('description', classValidationRules.description)}
+          />
+        </InputField>
+
+        {fields.map((field, index) => {
+          return (
+            <Classworks key={field.id}>
+              <Input
+                placeholder="name"
+                {...register(`classworks.${index}.name` as const, {
+                  required: true,
+                })}
+                defaultValue={field.name}
+              />
+              <Input
+                placeholder="description"
+                {...register(`classworks.${index}.description` as const, {
+                  required: true,
+                })}
+                defaultValue={field.description}
+              />
+
+              <TrashIcon type="button" onClick={() => remove(index)} />
+            </Classworks>
+          );
+        })}
+        <PrimaryButton
+          type="button"
+          onClick={() =>
+            append({
+              name: '',
+              description: '',
+            })
+          }
+        >
+          <FaPlus /> New Class Work
+        </PrimaryButton>
+      </StyledForm>
+      {Object.keys(errors).length > 0 && <FormErrorMessages errors={errors} />}
```

Comment:
do you want to try, instead of this, wrap each of the field with the error message? See the `EditorPlanModal` for example

```
<div>
          <InputField htmlFor="description">
            <span>Plan Description</span>
            <TextArea
              id="description"
              placeholder="Enter some description about the plan"
              {...register('description', planValidationRules.description)}
              hasError={Boolean(formErrors.description?.message)}
            />
          </InputField>
          {formErrors.description?.message ? (
            <StyledErrorMessage>
              {formErrors.description?.message}
            </StyledErrorMessage>
          ) : null}
        </div>
```